gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2424,8 +2670,18 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
             nodata.nVal64 = NC_FILL_INT64;
         }
 #endif
-        status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
-                            &nVarID);
+
+        if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation